Add Goal and User entities

NahomAlemu opened the pull request:
- Include controllers for both entities.
- Create repos for CRUD operations.
- Implement service layers.
- Add necessary dependencies.
- Establish database connectivity and add environment variable for DB credentials.

NahomAlemu committed:
Add Goal and User entities
- Include controllers for both entities.
- Create repos for CRUD operations.
- Implement service layers.
- Add necessary dependencies.
- Establish database connectivity and add environment variable for DB credentials.

zenwattage reviewed (approved):
GTG

CourtneyHoppus reviewed (approved):
I reviewed the file changes that include adding dependencies for our project in the pom.xml, some commenting on the application.java, and adding some application properties in that file for database connection. I did pull this code or run the app.
Everything looks okay to me.

NahomAlemu committed:
Add Goal and User entities
- Include controllers for both entities.
- Create repos for CRUD operations.
- Implement service layers.
- Add necessary dependencies.
- Establish database connectivity and add environment variable for DB credentials.

NahomAlemu committed:
fixed build fail - added env.properties

NahomAlemu commented:
Approval from Senthil is pending, but the code is merged for downloading purposes.

senthil1288 commented:
Hi Nahom,

I ll take a look at the PR and let you know if there are any comments
offline.  Thanks for merging.

On Mon, Aug 7, 2023, 11:49 AM NahomAlemu ***@***.***> wrote:

> Closed #1
> <https://github.com/CodeDay-ProductivityApp/codeDay-productivity/pull/1>.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/CodeDay-ProductivityApp/codeDay-productivity/pull/1#event-10027690243>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BBVCVK32GZDC3GIYKPNBSWTXUE2CTANCNFSM6AAAAAA27BS6QI>
> .
> You are receiving this because your review was requested.Message ID:
> <CodeDay-ProductivityApp/codeDay-productivity/pull/1/issue_event/10027690243
> @github.com>
>

senthil1288 commented on pom.xml:
```diff
@@ -21,7 +21,25 @@
 			<groupId>org.springframework.boot</groupId>
 			<artifactId>spring-boot-starter-web</artifactId>
 		</dependency>
-
```
You might also want to add a logging framework like log4j.

senthil1288 commented on pom.xml:
```diff
@@ -21,7 +21,25 @@
+			<artifactId>snakeyaml</artifactId>
+			<version>2.0</version>
+		</dependency>
+		<dependency>
```
How about a testing framework like junit or junit5?

NahomAlemu replied:
The transitive version of a dependency might have known bugs or issues. I had to explicitly add this dependency because I got a warning/error in the **spring-boot-starter** dependency while the build ran in the GitHub action. So to avoid problems, I had to add the dependency and control the version.

senthil1288 commented on src/main/java/com/codeday/productivity/entity/Goal.java:
```diff
@@ -0,0 +1,55 @@
+    private Instant startDate;
+
+    @Column(columnDefinition = "TIMESTAMP")
+    private Instant endDate;
```
Is this nullable?  We might want to keep it nullable, as goals could potentially not have an end data.

NahomAlemu replied:
It is nullable by default since I have not annotated it to be non-nullable. I thought goal would have an end date.

senthil1288 commented on src/main/java/com/codeday/productivity/entity/Goal.java:
```diff
@@ -0,0 +1,55 @@
+    @Column(columnDefinition = "TIMESTAMP")
+    private Instant endDate;
+
+    @Column(name = "due_date", columnDefinition = "TIMESTAMP")
```
How is due date different from end date?

NahomAlemu replied:
We made this design change in our discussion with the team. Due date indicates when the goal is expected to be completed, while the end date signifies the actual date the goal was concluded or terminated.

senthil1288 commented on src/main/java/com/codeday/productivity/controller/UserController.java:
```diff
@@ -0,0 +1,66 @@
+
+@RestController
+@RequestMapping("/v1")
+public class UserController {
```
Also, you should start adding unit tests for these functionalities.  In general, you should think about following test driven development approach so as to write unit tests as the same time as feature code or before writing feature code.
